### DISCUSSION TITLE
Why useContractRead returning object with hex instead of number?

### DISCUSSION THREAD

**Discussion by MubashirWaheed** [wevm/wagmi#1421] [Q&A] [answered]
I am trying to read the contract balance using `balanceOf` function  
here is code 
```
const { data, isLoading, error } = useContractRead({
    // contract address
    address: "0x53F2d86EDcb67C9eA04eaDa4C2333aeBeaB717eF",
    abi: [
      {
        inputs: [
          {
            internalType: "address",
            name: "owner",
            type: "address",
          },
        ],
        name: "balanceOf",
        outputs: [
          {
            internalType: "uint256",
            name: "",
            type: "uint256",
          },
        ],
        stateMutability: "view",
        type: "function",
      },
    ],
    functionName: "balanceOf",
    // contract deployer address
    args: ["0x3Ab95173E0E50469B200Ec44562Ca166d6B6a3f3"],
  });
```
instead of getting a number as data, I am getting this object
```js
{
_hex : "0x0b"
_isBigNumber: true
}
```
I tried converting `_hex` to string but it's not working
What exactly is the issue? 
NB: contract is not empty

**Comment by matthewlilley** [comment, marked as answer]
It's returning ethers BigNumber

https://docs.ethers.io/v5/api/utils/bignumber/

**Reply by MubashirWaheed** [reply]
Yes you are correct and I solved it by converting it to` ethers.utils.formatEther(data._hex)`

**Reply by MubashirWaheed** [reply]
But I am not getting the correct balance. The contract has `0.01202 Ether` but I am getting `0.000011`. I converted it to 6 digits by `ethers.utils.formatUnits(data._hex, 6)`
You can check the contract balance here: https://goerli.etherscan.io/address/0x53F2d86EDcb67C9eA04eaDa4C2333aeBeaB717eF#code

**Reply by MubashirWaheed** [reply]
I had other bugs in my code
